src/utils/rw_spin_lock.hpp

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+#pragma once
+
+#include <time.h>
```

Comment:
is there a c++ library for this?

Reply:
ctime?

Reply:
nanosleep is not part of the C standard, using time.h as a deliberate indication of non-portability.